app/src/test/java/com/kickstarter/viewmodels/CampaignDetailsViewModelTest.kt

```diff
@@ -3,24 +3,98 @@ package com.kickstarter.viewmodels
 
+    private val goBackToProject = TestSubscriber<Void>()
+    private val pledgeContainerIsVisible = TestSubscriber<Boolean>()
     private val url = TestSubscriber<String>()
 
-    private fun setUpEnvironment(environment: Environment, project: Project) {
+    private fun setUpEnvironment(environment: Environment, projectData: ProjectData) {
         this.vm = CampaignDetailsViewModel.ViewModel(environment)
 
+        this.vm.outputs.goBackToProject().subscribe(this.goBackToProject)
+        this.vm.outputs.pledgeContainerIsVisible().subscribe(this.pledgeContainerIsVisible)
         this.vm.outputs.url().subscribe(this.url)
 
-        // Configure the view model with a project.
-        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, project))
+        // Configure the view model with ProjectData.
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT_DATA, projectData))
+    }
+
+    @Test
+    fun testGoBackToProject() {
+        setUpEnvironment(environment(), ProjectDataFactory.project(ProjectFactory.project()))
+
+        this.vm.inputs.pledgeActionButtonClicked()
+        this.goBackToProject.assertValueCount(1)
+    }
+
+    @Test
+    fun testPledgeContainerIsVisible_whenProjectIsLiveNotBacked_control() {
+        setUpEnvironment(environment(), ProjectDataFactory.project(ProjectFactory.project()))
+
+        this.vm.inputs.pledgeActionButtonClicked()
+        this.pledgeContainerIsVisible.assertValue(false)
+    }
+
+    @Test
+    fun testPledgeContainerIsVisible_whenProjectIsLiveNotBacked_variant1() {
+        val environment = environment()
+                .toBuilder()
+                .optimizely(object : MockExperimentsClientType(){
+                    override fun variant(experiment: OptimizelyExperiment.Key, user: User?, refTag: RefTag?): OptimizelyExperiment.Variant {
+                        return OptimizelyExperiment.Variant.VARIANT_1
+                    }
+                })
+                .build()
+        setUpEnvironment(environment, ProjectDataFactory.project(ProjectFactory.project()))
+
+        this.vm.inputs.pledgeActionButtonClicked()
+        this.pledgeContainerIsVisible.assertValue(false)
+    }
+
+    @Test
+    fun testPledgeContainerIsVisible_whenProjectIsLiveNotBacked_variant2() {
+        val environment = environment()
+                .toBuilder()
+                .optimizely(object : MockExperimentsClientType(){
+                    override fun variant(experiment: OptimizelyExperiment.Key, user: User?, refTag: RefTag?): OptimizelyExperiment.Variant {
+                        return OptimizelyExperiment.Variant.VARIANT_2
+                    }
+                })
+                .build()
+        setUpEnvironment(environment, ProjectDataFactory.project(ProjectFactory.project()))
+
+        this.vm.inputs.pledgeActionButtonClicked()
+        this.pledgeContainerIsVisible.assertValue(true)
+    }
+
+    @Test
+    fun testPledgeContainerIsVisible_whenProjectIsNotLive() {
```

Comment:
For these last two I'm guessing the container is not visible by default but not sure if we want to explicitly emit `false` in these cases 🤷‍♂

Reply:
It is initially hidden but I can make it emit in those cases 👍